### ISSUE TITLE
Distribution list with over 50 members causes Invoke-RsCloudOfficeRequest to hang

### ISSUE THREAD

**Issue by matthooper** [rackerlabs/Invoke-RsCloudOfficeRequest#3]
To reproduce, fetch a list of members for a distribution list that has over 50 members. For example: 
`Invoke-RsCloudOfficeRequest "/v1/domains/example.com/ex/distributionlists/ExampleDL/members"`

The script never returns but repeatedly tries to download the first or second page. I notice that adding page and offset querystring parameters don't affect the results in any way.

**Comment by mkropat** [comment]
Thanks for the bug report. I'll try to reproduce it and investigate the cause sometime this week.

**Comment by matthooper** [comment]
Thanks Michael! Let me know if you’d like me to test a fix.

**Comment by mkropat** [comment]
Are you trying to use the `-UnpaginateProperty ...` feature by any chance? Unfortunately, the paging logic built into `Invoke-RsCloudOfficeRequest` does not currently support the Exchange endpoints (including the distribution list endpoints) [because those endpoints use a different paging format](http://api-wiki.apps.rackspace.com/api-wiki/index.php/Common_Exchange_API_Concepts#Listings).

Until that functionality is added to `Invoke-RsCloudOfficeRequest`, you can get all distribution list members with the following code:

```powershell
$result = @()
$marker = ''
do {
    $page = Invoke-RsCloudOfficeRequest "/v2/domains/example.com/ex/distributionlists/ExampleDL/members?marker=$marker"
    $result = $result + ($page.Recipients | select -ExpandProperty Value)

    $marker = $result[-1]
} while ($result.Length -lt $page.Total)

$result
```

**Comment by mkropat** [comment]
I've added a new feature request / issue, #4, about supporting de-paginating the Exchange API endpoints. I think that it would be a great feature to add. Currently I don't have any schedule about when such a feature would be implemented though.

**Comment by matthooper** [comment]
Yes, I’m using –UnpaginateProperty. Good catch and thanks for providing the code below!
